examples/nlp/language_modeling/tuning/megatron_gpt_qat.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch.multiprocessing as mp
+from omegaconf.omegaconf import OmegaConf
+from tqdm import tqdm
+
+from nemo.collections.nlp.models.language_modeling.megatron_gpt_sft_model import MegatronGPTSFTModel
+from nemo.collections.nlp.parts.megatron_trainer_builder import MegatronLMPPTrainerBuilder
+from nemo.core.config import hydra_runner
+from nemo.export.quantize import Quantizer
+from nemo.utils import logging
+from nemo.utils.exp_manager import exp_manager
+
+mp.set_start_method("spawn", force=True)
+
+"""
+This is a modified version of `megatron_gpt_finetuning.py` to perform PTQ and QAT on a SFT Model like Llama2-7b.
+Please see docs/source/nlp/quantization.rst for more details on the usage.
+"""
+
+
+def get_forward_loop(dataloader, num_batches):
+    if len(dataloader) < num_batches:
+        logging.warning(
+            f"Dataloader has fewer batches ({len(dataloader)}) than required ({num_batches}) for calibration."
+        )
+
+    def forward_loop(model):
+        data_iter = iter(dataloader)
```

Comment:
you could maybe use `https://docs.python.org/3/library/itertools.html#itertools.islice` for dataloader to avoid that `i >= num_batches` 2 lines later as a minor simplification